contrib/icinga2-commands.conf

```diff
@@ -8,81 +8,198 @@ template CheckCommand "cloud_aws" {
         }
         "-P" = {
             value= "$cloud_aws_profile$"
-            description = "AWS credential profile (~/.aws/credentials)"
+            description = "The AWS profile name, which represents a separate credential profile in the credential file"
         }
         "-R" = {
             value = "$cloud_aws_region$"
-            description = "AWS region name (e.g. eu-central-1)"
+            description = "The AWS region to send requests to"
+        }
+        "-C" = {
+            value= "$cloud_aws_credentials$"
+            description = "Path to the credentials file"
         }
     }
 }
 
 object CheckCommand "cloud_aws_instance" {
     import "cloud_aws"
 
+    command += [ "ec2", "instance" ]
+
     arguments += {
-        "ec2" = {
-            value = "$cloud_aws_ec2$"
-            description = "Checks in the EC2 context"
-        }
-        "instance" {
-            value = "$cloud_aws_instance$"
-            description = "Checks a single EC2 instance"
-        }
         "-i" = {
-            value = "$cloud_aws_id$"
+            value = "$cloud_aws_ec2_instance_id$"
             description = "Look for ec2 instance by id"
         }
         "-n" = {
-            value = "$cloud_aws_name$"
+            value = "$cloud_aws_ec2_instance_name$"
             description = "Look for ec2 instance by name"
         }
     }
-
-    vars.cloud_aws_ec2 = "ec2"
-    vars.loud_aws_instance = "instance"
 }
 
-object CheckCommand "cloud_aws_instance" {
+object CheckCommand "cloud_aws_instances" {
     import "cloud_aws"
 
+    command += [ "ec2", "instances" ]
+
     arguments += {
-        "ec2" = {
-            value = "$cloud_aws_ec2$"
-            description = "Checks in the EC2 context"
-        }
-        "instances" = {
-            value = "$cloud_aws_instances$"
-            description = "Checks multiple EC2 instances by name"
-        }
         "-n" = {
-            value = "$cloud_aws_name$"
+            value = "$cloud_aws_ec2_instances_name$"
             description = "Search for ec2 instances by name (e.g. instance*)"
         }
         "-a" = {
-            value = "$cloud_aws_autoscale$"
+            value = "$cloud_aws_ec2_instances_autoscale$"
             description = "Search for ec2 instances by autoscaling group"
         }
     }
+}
+
+object CheckCommand "cloud_aws_cloudfront" {
+    import "cloud_aws"
+
+    command += [ "cloudfront" ]
+
+    arguments += {
+        "-e" = {
+            value = "$cloud_aws_etag$"
+            description = "Etag name of one or multiple distributions. If --etag is empty, all distributions will be evaluated"
+        }
+    }
+}
+
+object CheckCommand "cloud_aws_s3_bucket" {
+    import "cloud_aws"
+
+    command += [ "s3", "bucket" ]
 
-    vars.cloud_aws_ec2 = "ec2"
-    vars.loud_aws_instances = "instances"
+    arguments += {
+        "-b" = {
+            value = "$cloud_aws_s3_bucket_buckets$"
```

Comment:
`cloud_aws_s3_bucket_buckets` is not a nice name, so while I understand the schema, repeating bucket seams unnecessary. Personal preference would be `cloud_aws_s3_buckets`.

Reply:
Done